eclair-core/src/main/scala/fr/acinq/eclair/payment/send/MultiPartPaymentLifecycle.scala

```diff
@@ -264,7 +264,7 @@ class MultiPartPaymentLifecycle(nodeParams: NodeParams, cfg: SendPaymentConfig,
           }
           paymentSent.feesPaid + localFees
       }
-      context.system.eventStream.publish(PathFindingExperimentMetrics(cfg.recipientAmount, fees, status, duration, now, isMultiPart = true, request.routeParams.experimentName, cfg.recipientNodeId))
+      context.system.eventStream.publish(PathFindingExperimentMetrics(cfg.paymentHash, cfg.recipientAmount, fees, status, duration, now, isMultiPart = true, request.routeParams.experimentName, cfg.recipientNodeId, cfg.invoice.map(_.routingInfo.flatMap(_.headOption.map(_.nodeId)).toSet).getOrElse(Set.empty)))
```

Comment:
This snippet is obscure enough to have to be put on a separate line with a name that gives a hint (pun intended) to the reader as to what it is.

```suggestion
      val firstHopOfRouteHints = cfg.invoice
        .map(_.routingInfo).getOrElse(Seq.empty)
        .flatMap(_.headOption.map(_.nodeId)).toSet
      context.system.eventStream.publish(PathFindingExperimentMetrics(cfg.paymentHash, cfg.recipientAmount, fees, status, duration, now, isMultiPart = true, request.routeParams.experimentName, cfg.recipientNodeId, firstHopOfRouteHints))
```